src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -585,77 +589,93 @@ discover_daos_mount(void)
 	if (fs_root == NULL)
 		return;
 
+	if (num_dfs >= MAX_DAOS_MT) {
+		D_WARN("dfs_list[] is full already. Need to incease MAX_DAOS_MT.\n");
+		return;
+	}
+
+	if (access(fs_root, R_OK)) {
+		D_DEBUG(DB_ANY, "no read permission for %s: %d (%s)\n", fs_root, errno,
+			strerror(errno));
+		return;
+	}
+
 	idx = query_dfs_mount(fs_root);
 	if (idx >= 0)
 		return;
 
-	/* daos_init() is not called yet, so we can not use D_WARN() etc. */
 	/* Not found in existing list, then append this new mount point. */
 	len_fs_root = strnlen(fs_root, DFS_MAX_PATH);
 	if (len_fs_root >= DFS_MAX_PATH) {
-		D_WARN("DAOS_MOUNT_POINT is too long. It is ignored.");
+		D_DEBUG(DB_ANY, "DAOS_MOUNT_POINT is too long. It is ignored.\n");
 		return;
 	}
 
 	pool = getenv("DAOS_POOL");
 	if (pool == NULL) {
-		D_WARN("warning> DAOS_POOL is not set.\n");
+		D_DEBUG(DB_ANY, "DAOS_POOL is not set.\n");
 		return;
 	}
 
 	container = getenv("DAOS_CONTAINER");
 	if (container == NULL) {
-		D_WARN("warning> DAOS_CONTAINER is not set.\n");
+		D_DEBUG(DB_ANY, "DAOS_CONTAINER is not set.\n");
 		return;
 	}
 
-	pt_end = stpncpy(dfs_list[num_dfs].fs_root, fs_root, DFS_MAX_PATH - 1);
-	if ((long int)(pt_end - dfs_list[num_dfs].fs_root) >= (DFS_MAX_PATH - 1)) {
-		D_WARN("dfs_list[num_dfs].fs_root[] is not large enough.\nQuit\n");
+	D_STRNDUP(dfs_list[num_dfs].fs_root, fs_root, len_fs_root);
+	if (dfs_list[num_dfs].fs_root == NULL)
```

Comment:
It's unlikely, but possible that this fails with `ENOMEM`. Do we need more explicit error handling for `discover_daos_mount`?

Reply:
https://github.com/daos-stack/daos/pull/12259 already adds a test (which fails) for error handling on this very line of code however.

Reply:
ok. I will return error code for discover_dfuse() and discover_daos_mount() to address this issue. Just create a ticket, https://daosio.atlassian.net/browse/DAOS-13574
Thank you!